apps/docs/build.md

```diff
@@ -172,49 +172,75 @@ different configurations get run. Here is the approximate flow of commands/files
 
    The rails asset pipeline can be configured to do bundling and minification,
    but it is a significantly less powerful tool than webpack. As a result, we
-   only use the rails asset pipeline to add "digests" to the filenames. In
-   effect, the `assets:precompile` rake task just copies over all the files that
-   were generated by webpack, adding a unique digest hash to the
-   filenames, and constructing a manifest file with the mapping. You can find
-   all these files in `dashboard/public/assets/`.
-
-   When loading a bundle with a script tag inside an erb or haml
-   file, rails will automatically add the unique digest to the url by looking at
-   the manifest file generated during precompilation. The manifest file is
-   stored at
+   only use the rails asset pipeline to add "digests" to the few assets which
+   do not already contain hashes generated by webpack, and to retain each
+   hashed/digested asset for a few deploys so that links to them do not break
+   immediately after each deploy.
+   
+   More specifically, the `assets:precompile` rake task does the following:
+   
+    * copies over all the files that were generated by webpack
+    * generates application.js and application.css and a copies over a few other 
+    non-webpack files with a unique digest added to the filenames
+    * constructs a sprockets manifest file which is used only for looking up non-webpack assets.
+     
+   You can find all these files in `dashboard/public/assets/`.
+   
+   When loading a bundle inside an erb or haml file via `asset_path`,
+   `stylesheet_link_tag`, or `javascript_include_tag`, rails will automatically
+   add the unique digest to the url by looking at the sprockets manifest file
+   generated during precompilation. The sprockets manifest file is stored at
    `dashboard/public/assets/.sprockets-manifest-<some-unique-hash>.json`
 
+   When loading a bundle in dashboard or pegasus in production via the
+   `webpack_asset_path` helper, the helper will look in the webpack manifest to
+   determine the full path of the asset to load, including the content hash.
+   This method is preferred over those methods listed in the previous paragraph.
```

Comment:
Does this mean that this method is interchangeable with the previous methods? Is it preferred because the previous methods are deprecated and will be eventually removed? Does it depend on where and how the asset is being stored / generated?

Also, is there a difference between 'bundle' and 'asset'? They are used to refer to the same thing in this paragraph. Is an image, for example, both a 'bundle' and an 'asset', or is it an 'asset' but not a 'bundle'? Does this preferred approach only work for 'bundles', or for all 'assets'?

Reply:
`webpack_asset_path` is not interchangeable with previous methods. Previous methods (`javascript_include_tag`, `stylesheet_link_tag` and `asset_path`) are deprecated, except for references to application.js / application.css.

a "bundle" is a file which was output specifically by webpack as defined here: https://webpack.js.org/concepts/#output

the [rails asset pipeline](https://edgeguides.rubyonrails.org/asset_pipeline.html) seems to refer to both its inputs and its outputs as "assets", and this is how I'd recommend that we use this term. 

Please let me know if that clarifies things for you. I will go through and make another pass of this document to follow (and perhaps define) this particular part of the terminology.

Reply:
Thanks, that helps quite a bit.

That said, maybe it's because I'm not directly familiar with how we currently use Webpack loaders, but how we're using the concept of 'bundle' (more specifically, what kinds of content we imply by our use of the term) still isn't quite clear to me. In earlier JS-pipeline frameworks, a 'bundle' was strictly a set of compiled JavaScript sources that would be executed as a self-contained unit, through a well-defined interface of entry-module exports. Now with lazy-loading, code-splitting, and Webpack's whole giant ecosystem of loader plugins, a Webpack 'bundle' can contain pretty much any kind of content you want, but sometimes it can still means 'mostly javascript, with a sprinkle of related content'.

I'd like a bit more clarity on what kinds of content we reasonably expect to load in webpack bundles (if it's more than strictly JS sources) - especially if this pipeline will be our 'preferred' approach for incorporating assets into our application moving forward.

Reply:
I've removed the reference to the term "bundle" from this part of my added comments. there are a few uses of this term earlier on the page, but I'd like to postpone tackling updates to those section because our bundling strategy is likely to change in the course of the upcoming optimizations.